poulpe-model/src/test/java/org/jtalks/poulpe/model/entity/PoulpeUserTest.java

```diff
@@ -31,4 +41,39 @@ public Group groupWithId(long id) {
         return group;
     }
 
+    @Test
+    public void theEntityFieldsShouldBeSerialized() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        PoulpeUser user = new PoulpeUser("username","email","password","salt");
+        user.setId(1);
+        user.setAvatar(new byte[]{1});
+        user.setVersion(1L);
+        user.setFirstName("First Name");
+        user.setLastName("Last Name");
+        user.setBanReason("Ban Reason");
+        user.setEnabled(true);
+        user.setRole("Role");
+        Method setLastLogin = User.class.getDeclaredMethod("setLastLogin", DateTime.class);
+        Method setEncodedUsername = User.class.getDeclaredMethod("setEncodedUsername", String.class);
+        setLastLogin.setAccessible(true);
+        setEncodedUsername.setAccessible(true);
+        setLastLogin.invoke(user,DateTime.now());
+        setEncodedUsername.invoke(user,"Encoded Username");
+
+        byte[] serialize = SerializationUtils.serialize(user);
+        PoulpeUser serializedUser = (PoulpeUser)SerializationUtils.deserialize(serialize);
+        assertReflectionEquals(user, serializedUser);
+    }
+
+    @Test
+    public void groupsFieldIsNotSerializable(){
+        List<Group> groups = PoulpeGroup.createGroupsWithNames("The Group");
+        List<Group> serializedGroups = new ArrayList<Group>();
+        try {
+            byte[] serialize = SerializationUtils.serialize(groups);
+            serializedGroups = (List<Group>) SerializationUtils.deserialize(serialize);
+        } catch (Exception e){
```

Comment:
In any case, if we invoke the test method with non-empty groups, exception has been thrown during serialization. And it's strange, because we have not any such exceptions during application executions. Moreover if trying to model the same situation with simple classes design serialization exceptions not happend. I think there is need to investigate this behaviour firstly and I'll try.